### ISSUE TITLE
failed to asynchronously prepare wasm: CompileError : WebAssembly.instantiate():expected 5201903 bytes, fell off end @+13

### ISSUE THREAD

**Issue by yangfangfang1204** [emscripten-core/emscripten#15537]
CompileError : WebAssembly.instantiate():expected 5201903 bytes, fell off end @+13

uncaught (in promise) abort ({}) at Error at jsStackTrace , if this abort() is unexpected, build with -s ASSERTIONS=1 which can give more information

**Comment by sbc100** [comment]
I looks like there could be some issue with the wasm file being delivered to your browser.

Is your wasm file really 5201903 bytes?   How are you delivering the wasm file?  Is it being fetched directly from the server?   Can you verify the wasm file being downloaded is valid?  Can you download it directly can validate it using `wasm-validate` from wabt, or `wasm-dis` from binaryen.

Can you share you link command? (you aren't using `SINGLE_FILE=1` or `WASM=0` are you?)

**Comment by yangfangfang1204** [comment]
yes , i am not using `SINGLE_FILE=1` or `WASM =0` , and i don't know how the user delivering the wasm file ,because there is lots of users who use the wasm library , we just generate the .js and .wasm library .

so i could use `SINGLE_FILE=1` or `WASM=0` to solve the problem ?

**Comment by sbc100** [comment]
No, I as not suggesting with `SINGLE_FILE` or `WASM=0`.    

Does the number of bytes in the error message match the size of your wasm file: 5201903?

Are you able to debug this issue in devtools (i.e. can you reproduce it yourself)?   It seems there is some kind of network issue or bad files being served.

**Comment by yangfangfang1204** [comment]
i can't reproduce it by myself , so we don't the users how to delivered the wasm library , maybe the server exist some error .

**Comment by yangfangfang1204** [comment]
there is another question :
`still waiting on run dependencies: dependency: wasm-instantiate , end of list`
